scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,567 @@
+get_kubeconfig() {
+    context="$1" # context in the kubeconfig
+    cp_name="$2" # name of the Control Plane
+    cp_type="$3" # type of the Control Plane
+
+    # wait for CP ready
+    while [[ $(k --context $context get controlplane $cp_name -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do
+        sleep 5
+    done
+
+    # put into the shell variable "kubeconfig" the kubeconfig contents for use from outside of the hosting cluster
+    if [[ "$cp_type" == "host" ]] ; then
+        kubeconfig=$(k --context $context config view --flatten --minify)
+    else
+        # determine the secret name and namespace
+                     key=$(k --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.key}')
+             secret_name=$(k --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.name}')
+        secret_namespace=$(k --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.namespace}')
+        # get the kubeconfig
+        kubeconfig=$(k --context $context get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}" | base64 -d)
+    fi
+
+    # return the kubeconfig file contents in YAML format
+    echo "$kubeconfig"
+}
+
+
+###############################################################################
+# Parse command line arguments
+###############################################################################
+while (( $# > 0 )); do
+    case "$1" in
+    (--kubeconfig|-K)
+        if (( $# > 1 ));
+        then { arg_kubeconfig="$2"; shift; }
+        else { echo "$0: missing kubeconfig filename" >&2; exit 1; }
+        fi;;
+    (--context|-C)
+        if (( $# > 1 ));
+        then { arg_context="$2"; shift; }
+        else { echo "$0: missing context name" >&2; exit 1; }
+        fi;;
+    (--logs|-L)
+        arg_logs=true;;
+    (--yaml|-Y)
+        arg_yaml=true;;
+    (--verbose|-V)
+        arg_verbose=true;;
+    (--version|-v)
+        echo "${SCRIPT_NAME} v${SCRIPT_VERSION}"
+        exit 0;;
+    (-X)
+        set -x;;
+    (-h|--help)
+        display_help
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+
+###############################################################################
+# Define alias definition
+###############################################################################
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+# Define the special kubectl
+k() {
+    KUBECONFIG="$kubeconfig" kubectl $*
+}
+
+
+###############################################################################
+# Script info
+###############################################################################
+echov -e "${COLOR_INFO}$($0 --version)${COLOR_NONE}\n"
```

Comment:
Can this script be used by piping it into `bash`? If so then `$0` will refer to `bash` rather than the script.

Reply:
I changed to ``${BASH_SOURCE[0]}`

Reply:
I don't think that is much of an improvement. When a user does `curl $something | bash`, that invocation of `bash` knows nothing about the URL that stdin was fetched from.

Reply:
ok, I changed to
```
echov -e "${COLOR_INFO}${SCRIPT_NAME} v${SCRIPT_VERSION}{COLOR_NONE}\n"
```